src/pyinfraboxutils/coverage.py

```diff
@@ -232,5 +286,29 @@ def __create_markup(self, badge_dir):
         return doc
 
     def parse(self, badge_dir):
-        self.__convert_xml()
+        if(self.input == "/"):
+            self.parseDir(badge_dir)
+        else:
+            self.__convert_xml()
         return self.__create_markup(badge_dir)
+
+    def parseDir(self, badge_dir):
+        for root, dirs, files in os.walk("."):
+            for filename in files :
+                if filename.endswith(".xml"):
+                    self.input = filename
+                    self.__convert_xml()
+
+        tmp_files = []
+        while self.files != []:
+            f1 = self.files.pop(0)
+
+            for f in self.files:
+                if f.name == f1.name:
+                    f1 = f1 + f
+                    self.files.remove(f)
+
+            tmp_files.append(f1)
+
+        self.files = tmp_files
+        self.input = "/"
```

Comment:
this is probably not needed and could be wrong

Reply:
What can wrong with it ? The loop merge File in a temporary list until the self.files is empty and then reverse them. Should I use an other way to merge file ?